sim/druid/lacerate.go

```diff
@@ -0,0 +1,121 @@
+package druid
+
+import (
+	"time"
+
+	"github.com/wowsims/sod/sim/core"
+	"github.com/wowsims/sod/sim/core/proto"
+)
+
+func (druid *Druid) registerLacerateSpell() {
+	if !druid.HasRune(proto.DruidRune_RuneLegsLacerate) {
+		return
+	}
+	initialDamageMul := 1.0
+
+	switch druid.Ranged().ID {
+	case IdolOfCruelty:
+		initialDamageMul += .07
+	}
+	rageMetrics := druid.NewRageMetrics(core.ActionID{SpellID: 431446})
+
+	druid.Lacerate = druid.RegisterSpell(Bear, core.SpellConfig{
+		ActionID:    core.ActionID{SpellID: 414644},
+		SpellSchool: core.SpellSchoolPhysical,
+		DefenseType: core.DefenseTypeMelee,
+		ProcMask:    core.ProcMaskMeleeMHSpecial,
+		Flags:       SpellFlagOmen | core.SpellFlagMeleeMetrics | core.SpellFlagAPL,
+
+		RageCost: core.RageCostOptions{
+			Cost:   core.TernaryFloat64(druid.BerserkAura.IsActive(), 0, 10),
+			Refund: 0.8,
+		},
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				GCD: core.GCDDefault,
+			},
+			IgnoreHaste: true,
+		},
+
+		DamageMultiplier: initialDamageMul,
+		ThreatMultiplier: 3.25,
+		// TODO: Berserk 3 target lacerate cleave - Saeyon
+
+		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+			baseDamage := (spell.Unit.MHWeaponDamage(sim, spell.MeleeAttackPower()) * .2) * float64(druid.LacerateBleed.Dot(target).GetStacks())
+
+			spell.DamageMultiplier = initialDamageMul
+			result := spell.CalcAndDealDamage(sim, target, baseDamage, spell.OutcomeMeleeSpecialHitAndCrit)
+
+			if result.Landed() {
+				druid.LacerateBleed.Cast(sim, target)
+
+				if druid.HasRune(proto.DruidRune_RuneHelmGore) && sim.Proc(0.15, "Gore") {
```

Comment:
Rune checks are really expensive and can be cached as variables, so can perform the rune check at the top of the function and assign it to something like `hasGoreRune`?